datafusion/core/tests/parquet/arrow_statistics.rs

```diff
@@ -614,6 +614,94 @@ async fn test_int_8() {
     .run();
 }
 
+#[tokio::test]
```

Comment:
I think this test is mostly a duplicate of `test_float16` just that one has `Check::RowGroup` vs ``Check::Both`... maybe one or the other can be updated?

Reply:
@tshauck this is one of the things that were bugging me a bit, the other one being https://github.com/apache/datafusion/pull/10982/files#diff-7110f4709c105a18ef74a212396444d62052179a735d148fb62470a8b157fb40R749-R763 -- both are very repetitive

however, I didn't want to get overeager, only to realize later than the abstractions chosen was not the right one. Perhaps the best way forward would be to address https://github.com/apache/datafusion/issues/10952 first (which may also have its own "float16"-like tricky case), and then getting the correct macros for testing, index handling, etc. Speaking of that, would you like to take that one? I'd be happy to review then

Reply:
Sounds like a good plan to me -- let's merge this PR and then work on https://github.com/apache/datafusion/issues/10952

Reply:
Filed https://github.com/apache/datafusion/issues/11000 to track duplication